src/VisualStudio/Core/Def/Implementation/ProjectTelemetry/VisualStudioProjectTelemetryService.cs

```diff
@@ -53,23 +54,35 @@ internal class VisualStudioProjectTelemetryService
         /// <summary>
         /// Queue where we enqueue the information we get from OOP to process in batch in the future.
         /// </summary>
-        private AsyncBatchingWorkQueue<ProjectTelemetryData>? _workQueue;
+        private readonly AsyncBatchingWorkQueue<ProjectTelemetryData>? _workQueue;
 
         [ImportingConstructor]
         [Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
-        public VisualStudioProjectTelemetryService(VisualStudioWorkspaceImpl workspace, IThreadingContext threadingContext) : base(threadingContext)
-            => _workspace = workspace;
+        public VisualStudioProjectTelemetryService(
+            VisualStudioWorkspaceImpl workspace,
+            IThreadingContext threadingContext) : base(threadingContext)
+        {
+            _workspace = workspace;
 
-        void IVisualStudioProjectTelemetryService.Start(CancellationToken cancellationToken)
-            => _ = StartAsync(cancellationToken);
+            _workQueue = new AsyncBatchingWorkQueue<ProjectTelemetryData>(
+                TimeSpan.FromSeconds(1),
+                NotifyTelemetryServiceAsync,
+                threadingContext.DisposalToken);
+        }
 
-        private async Task StartAsync(CancellationToken cancellationToken)
+        void IEventListener<object>.StartListening(Workspace workspace, object _)
+        {
+            if (workspace is VisualStudioWorkspace)
+                _ = StartAsync();
+        }
+
+        private async Task StartAsync()
         {
             // Have to catch all exceptions coming through here as this is called from a
             // fire-and-forget method and we want to make sure nothing leaks out.
             try
             {
-                await StartWorkerAsync(cancellationToken).ConfigureAwait(false);
+                await StartWorkerAsync().ConfigureAwait(false);
```

Comment:
💭 Should we wait for solution to fully load before starting this worker?

Reply:
yes we probably should i can add that after i get my other PR in.

Reply:
The other PR doesn't impact the call though, right?